src/main/java/net/vulkanmod/gl/GlTexture.java

```diff
@@ -35,11 +36,26 @@ public static int genTextureId() {
         return id;
     }
 
+    // Default texture can vary depending on graphic drivers,
+    // but it's usually a square that is either white or black
+    private static GlTexture getDefaultTexture() {
+        if (GlTexture.defaultTexture != null)
+            return GlTexture.defaultTexture;
+        GlTexture defaultTexture = new GlTexture(0);
+        defaultTexture.vulkanImage = VulkanImage.createWhiteTexture();
+        GlTexture.defaultTexture = defaultTexture;
+        return defaultTexture;
+    }
+
     public static void bindTexture(int id) {
         boundTextureId = id;
-        boundTexture = map.get(id);
+        if (id == 0) {
```

Comment:
Why not adding it directly to the map instead of doing that check?

Reply:
I can do that, but I would have to add a direct `if ()` statement for `glDeleteTextures`, as the documentations says for ID  deleting texture should be a noop.

Also, I cannot move image initialization to the static initializer as `VulkanImage` need the static initialize of `GlTexture` for some reason.

Reply:
OpenGL Documentation for `glDeleteTextures`:
- https://registry.khronos.org/OpenGL-Refpages/gl4/html/glDeleteTextures.xhtml